tests/h/settings_test.py

```diff
@@ -98,3 +98,63 @@ def test_mandrill_settings():
 
 def test_mandrill_settings_unset():
     assert settings.mandrill_settings({}) is None
+
+
+class TestSettingsManager(object):
+    def test_does_not_emit_warnings_when_deprecated_setting_is_not_used(self, log):
+        settings_manager = settings.SettingsManager({}, {})
+        settings_manager.set('foo', 'FOO', deprecated_msg='what to do instead')
+
+        assert not log.warn.called
+
+    def test_emits_warnings_when_deprecated_setting_is_used(self, log):
```

Comment:
In test names I think `*_warns_*` (and `*_doesnt_warn_*`) would be clearer and shorter than `*_emits_warning_*`